libs/tdigest/src/test/java/org/elasticsearch/tdigest/TDigestTests.java

```diff
@@ -0,0 +1,859 @@
+        assertEquals(4, last.mean(), 0);
+        assertEquals(1.0, digest.cdf(digest.getMax() + 1e-9), 0);
+        assertEquals(1 - 0.5 / digest.size(), digest.cdf(digest.getMax()), 0);
+        assertEquals(1 - 1.0 / digest.size(), digest.cdf((digest.getMax() - 1e-9)), 1e-10);
+
+        assertEquals(4, digest.quantile(1), 0);
+        assertEquals(last.mean(), 4, 0);
+    }
+
+    /**
+     * The example from issue #167
+     */
+    public void testIssue167() {
+        TDigest d = factory(100).create();
+        for (int i = 0; i < 2; ++i) {
+            d.add(9000);
+        }
+        for (int i = 0; i < 11; ++i) {
+            d.add(3000);
+        }
+        for (int i = 0; i < 26; ++i) {
+            d.add(1000);
+        }
+
+        assertEquals(3000.0, d.quantile(0.90), 1e-5);
+        assertEquals(3600.0, d.quantile(0.95), 1e-5);
+        assertEquals(5880.0, d.quantile(0.96), 1e-5);
+        assertEquals(8160.0, d.quantile(0.97), 1e-5);
+        assertEquals(9000.0, d.quantile(0.98), 1e-5);
+        assertEquals(9000.0, d.quantile(1.00), 1e-5);
+    }
+
+    public void testSingleValue() {
+        Random rand = random();
+        final TDigest digest = factory().create();
+        final double value = rand.nextDouble() * 1000;
+        digest.add(value);
+        final double q = rand.nextDouble();
+        for (double qValue : new double[] { 0, q, 1 }) {
+            assertEquals(value, digest.quantile(qValue), 0.001f);
+        }
+    }
+
+    public void testFewValues() {
+        // When there are few values in the tree, quantiles should be exact
+        final TDigest digest = factory().create();
+        final Random r = random();
+        final int length = r.nextInt(10);
+        final List<Double> values = new ArrayList<>();
+        for (int i = 0; i < length; ++i) {
+            final double value;
+            if (i == 0 || r.nextBoolean()) {
+                value = r.nextDouble() * 100;
+            } else {
+                // introduce duplicates
+                value = values.get(i - 1);
+            }
+            digest.add(value);
+            values.add(value);
+        }
+        Collections.sort(values);
+
+        // for this value of the compression, the tree shouldn't have merged any node
+        assertEquals(digest.centroids().size(), values.size());
+        for (double q : new double[] { 0, 1e-10, r.nextDouble(), 0.5, 1 - 1e-10, 1 }) {
+            double q1 = Dist.quantile(q, values);
+            double q2 = digest.quantile(q);
+            assertEquals(String.format(Locale.ROOT, "At q=%g, expected %.2f vs %.2f", q, q1, q2), q1, q2, 0.03);
+        }
+    }
+
+    // private int repeats() {
+    // return Boolean.parseBoolean(System.getProperty("runSlowTests")) ? 10 : 1;
+    // }
+
+    public void testEmptyDigest() {
+        TDigest digest = factory().create();
+        assertEquals(0, digest.centroids().size());
+        assertEquals(0, digest.centroids().size());
+    }
+
+    // /**
```

Comment:
There are a lot of code commented out like this. Maybe just remove it or enable it?

Reply:
Indeed, removed.